internal/metarepos/storage.go

```diff
@@ -585,6 +593,49 @@ func (ms *MetadataStorage) updateLogStream(ls *varlogpb.LogStreamDescriptor) err
 	return err
 }
 
+func (ms *MetadataStorage) willBeSealed(ls *varlogpb.LogStreamDescriptor) bool {
+	pre, cur := ms.getStateMachine()
+
+	reports, ok := cur.LogStream.UncommitReports[ls.LogStreamID]
+	if !ok {
+		tmp, ok := pre.LogStream.UncommitReports[ls.LogStreamID]
+		if !ok {
+			return false
+		}
+
+		reports = proto.Clone(tmp).(*mrpb.LogStreamUncommitReports)
+	}
+
+	lastCommitted := ms.getLastCommittedLLSN(ls.TopicID, ls.LogStreamID)
+	for _, r := range ls.Replicas {
+		o := reports.Replicas[r.StorageNodeID]
+		if lastCommitted < o.UncommittedLLSNOffset {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (ms *MetadataStorage) getLastCommittedLLSN(topicID types.TopicID, lsID types.LogStreamID) types.LLSN {
+	crs := ms.GetLastCommitResults()
+	if crs == nil {
+		return types.InvalidLLSN
+	}
+
+	r, _, ok := crs.LookupCommitResult(topicID, lsID, -1)
+	if !ok {
+		// newbie
+		return types.InvalidLLSN
+	}
+
+	if r.CommittedLLSNOffset+types.LLSN(r.CommittedGLSNLength) == types.InvalidLLSN {
+		return types.InvalidLLSN
+	}
```

Comment:
FYI, since the offset should be greater than zero, and the length should be greater than or equal to zero, it is odd if this condition hits.

Reply:
It means InvalidLogStreamCommitResult, but as pointed out, it is an unreachable condition by L628.